File: maxhanna.client/src/services/datacontracts/file-comment.ts
import { User } from "./user";

export class FileComment {
  constructor(id: number, fileId: number, user: User, commentText: string, upvotes: number, downvotes: number) {
    this.id = id;
    this.fileId = fileId;
    this.user = user;
    this.commentText = commentText;
    this.upvotes = upvotes;
    this.downvotes = downvotes;
  }

  id: number;
  fileId: number;
  user: User;
  commentText: string; 
  upvotes: number;
  downvotes: number;
}

File: maxhanna.Server/Controllers/DataContracts/FileComment.cs
namespace maxhanna.Server.Controllers.DataContracts
{
    public class FileComment
    {
        public int Id { get; set; }
        public int? FileId { get; set; }
        public int? StoryId { get; set; }
        public User? User { get; set; }
        public string? CommentText { get; set; } 
        public DateTime Date { get; set; }
        public List<FileEntry>? CommentFiles { get; set; } = new List<FileEntry>();
        public List<Reaction>? Reactions { get; set; }
    }
}
